apps/backend/src/agents/companion-runner.ts

```diff
@@ -44,52 +79,99 @@ export class LangGraphResponseGenerator implements ResponseGenerator {
     }
   ) {}
 
-  async run(params: GenerateResponseParams): Promise<GenerateResponseResult> {
+  async run(params: GenerateResponseParams, callbacks: ResponseGeneratorCallbacks): Promise<GenerateResponseResult> {
     const { modelRegistry, checkpointer } = this.deps
-    const { threadId, modelId, systemPrompt, messages } = params
+    const { threadId, modelId, systemPrompt, messages, streamId, sessionId, personaId, lastProcessedSequence } = params
 
     logger.debug(
       {
         threadId,
         modelId,
         messageCount: messages.length,
+        streamId,
+        sessionId,
       },
       "Running companion graph"
     )
 
+    // Track messages sent for the tool
+    const sentMessageIds: string[] = []
+    let messagesSentCount = 0
```

Comment:
What happens if the agent process crashes?  I think we should persist exactly which messages the agent sent in a given session